### PR TITLE
Dashboard Personalization: Implements the Personalization Screen 

### LINKED ISSUE #18940: Dashboard Personalization: Create Personalize Home Tab view
Implement the Personalize Home Tab view

Design Reference: YSMw2nbFAgPpjZLLbYvTfT-fi-0_1
```[tasklist]
### Tasks
- [x] Create Fragment, VM, Compose UI (uiState, uiModel)
- [x] Hook in logic for saving hide/show preferences (separate task for implementing the logic) - https://github.com/wordpress-mobile/WordPress-Android/pull/19178
- [x] Track events - https://github.com/wordpress-mobile/WordPress-Android/pull/19183
- [x] Next Steps (For sites that select show me around) - https://github.com/wordpress-mobile/WordPress-Android/pull/19178
```

### LINKED ISSUE #18942: Dashboard Personalization: Implement logic to hide/show cards based on preferences set
Implement the logic to support hiding/showing _some_ dashboard cards based on user preference setting.

- This will probably have some architecture changes in MySiteViewModel and MySiteCardAndItem. Consider the more menu "hide this" action will need to toggle off the preference.
```[tasklist]
### Tasks
- [ ] Today's stats
- [ ] Draft posts
- [ ] Scheduled posts
- [ ] Blaze
- [ ] Blogging prompts
- [ ] Pages
- [ ] Recent Activity
- [ ] Next Steps (only if selected by user via the pop-up)
```

### PR DESCRIPTION
**Pull request by AjeshRPai**
## Issue
Part of #18940 and Closes #18942

## Description 
This PR implements the Personalization screen for dashboard cards

|  Light | Dark   | 
|---|---|
| Personalization_Screen_light  | personalization_screen_dark   | 

Note: the tracking will be taken up in a seperate PR 

## To test:
1. Go to Jetpack app 
2. Go to Me → Debug Settings → Enable `dashboard_personalization`
3. Go to the bottom of dashboard
4. Click on `Personalize your home tab`
5. ✅  Verify that the visibility of the cards are shown correct 
6. Toggle the visibility of any card 
7. Go to dashboard 
8. Verify that the card is hidden on Dashboard 
9. Repeat steps 6-8 for each card 

## Regression Notes
1. Potential unintended areas of impact
Dashboard cards are not hidden/shown properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and Unit testing

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by osullivanchris** [comment]
@AjeshRPai this is looking really good. I can't find any major issues right now but I'll keep using it.

I wouldn't consider it a must-have, but one thing we did on iOS that we haven't done on Android yet is to show a message if the user turns off all the cards.

**Comment by AjeshRPai** [comment]
Hey @osullivanchris 👋🏼 
>I wouldn't consider it a must-have, but one thing we did on iOS that we haven't done on Android yet is to show a message if the user turns off all the cards.

I will try to implement this before the release

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/personalization/PersonalizationActivity.kt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                        text = stringResource(id = R.string.personalization_screen_footer_cards),
+                        fontSize = 13.sp,
+                        fontWeight = FontWeight.Normal,
+                        color = MaterialTheme.colors.onSurface.copy(alpha = ContentAlpha.medium)
+                    )
+                }
+            }
+        }
+    }
+}
```
Comment (ravishanker) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/personalization/PersonalizationActivity.kt:111]: Please add a @Preview to the Composables, in your next PR.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/personalization/PersonalizationViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    @param:Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+    private val appPrefsWrapper: AppPrefsWrapper,
+    private val selectedSiteRepository: SelectedSiteRepository,
+    private val bloggingRemindersStore: BloggingRemindersStore,
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper
+) : ScopedViewModel(bgDispatcher) {
+    private val _uiState = MutableLiveData<List<DashboardCardState>>()
+    val uiState: LiveData<List<DashboardCardState>> = _uiState
+
+    fun start() {
```
Comment (ravishanker) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/personalization/PersonalizationViewModel.kt:33]: For some reason this start function in ViewModel was used in the past.  Not sure, if it is needed anymore.  We could probably use init() instead, then don't have to have call this in onCreate in Activity explicitly.

**Review by ravishanker — APPROVED** [review]
Thanks for pushing Compose journey 👍 

Left a couple of comments for your consideration in your next PR.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/personalization/PersonalizationViewModel.kt (hunk 2 above)
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/personalization/PersonalizationViewModel.kt:33]: Hey @ravishanker 
I initially used the `init()` function, but on test functions, it was becoming harder to mock all the items for a test. So if we use the `init` block, I have to either set up all the mocks in the @Before function of tests or initialize ViewModel only in tests.  Both solutions didn't look right to me somehow, hence resorted to using `start()`. 

If you have any suggestions, I am all ears.

**Pull request merged**